apps/dashboard/src/components/editor/use-autosave.js

```diff
@@ -62,9 +63,11 @@ export const useAutosave = ( projectId, editorView ) => {
 			// Clear any previous autosave timers
 			clearTimeout( autosave.current );
 
+			const title = getProjectTitle();
+
 			// Set a new timer
 			autosave.current = setTimeout( () => {
-				saveEditorContent( projectId, parse( content ) );
+				saveEditorContent( projectId, parse( content ), { title } );
```

Comment:
Not a huge fan of how this will append the current title value to all update requests. With this affecting stuff like permalinks etc.
Another issue is, editing the title now affects the 'unpublished changes' notice.

On top of that, I believe it's still susceptible to race conditions and the title only gets saved if you edit the content at some point afterwards.

I'm not sure how we can resolve the latter two issues without bumping the save timer logic up into the data layer so it can properly track to-be-saved data coming in from different sources.

Reply:
> Not a huge fan of how this will append the current title value to all update requests.

Why not?

> With this affecting stuff like permalinks etc.

That is a problem of backend and should be dealt with there. From the editor we just send the usual payload of data to be saved

> editing the title now affects the 'unpublished changes' notice

Yeah, didn't think about that 

> I believe it's still susceptible to race conditions...

Not sure "what" race conditions you're talking about.

> and the title only gets saved if you edit the content at some point afterwards.

I'll repeat this PR's intro: *I could say either I didn't find a good way or a good reason for this so, it's by design.*

> I'm not sure how we can resolve the latter two issues without bumping the save timer logic up into the data layer so it can properly track to-be-saved data coming in from different sources.

Well, if this is definitely the goal, then we can just move the things we want to our *own* state branch and do all this data layer dancing that you see fit (title, editorContent, timers/autosave/notices)

Reply:
It is, and it'll also help with stuff like having to fetch the content manually from the editor anytime you want to force a save.  
Will also be useful for stuff like paging for the same issue that now, if you would like to add a new page, you need to fetch the editor content manually because we're updating the entire thing and we don't want any debounced changes waiting to be saved to get lost when we switch context.